sdk/client-core/core/src/main/java/io/clientcore/core/http/models/HttpRequest.java

```diff
@@ -178,60 +191,70 @@ public HttpRequest setBody(BinaryData content) {
     }
 
     /**
-     * Get the request metadata.
+     * Get the request {@link RequestOptions options}.
      *
-     * @return The request metadata.
+     * @return The request {@link RequestOptions options}.
      */
-    public HttpRequestMetadata getMetadata() {
-        return metadata;
+    public RequestOptions getOptions() {
+        return options;
     }
 
     /**
-     * Set the request metadata.
+     * Set the request {@link RequestOptions options}.
      *
-     * @param metadata The request metadata.
+     * @param options The request {@link RequestOptions options}.
      *
      * @return The updated {@link HttpRequest}.
      */
-    public HttpRequest setMetadata(HttpRequestMetadata metadata) {
-        Objects.requireNonNull(metadata, "'metadata' cannot be null");
+    public HttpRequest setOptions(RequestOptions options) {
```

Comment:
I wonder if this should be `setRequestOptions` - I would lean towards calling it this, to help differentiate between some other options type.